content/en/blog/guestbook-application-with-fission-golang.md

```diff
@@ -0,0 +1,310 @@
+fission fn create --name restapi-post --executortype newdeploy --maxscale 3 --env go --pkg restapi-go-pkg --entrypoint MessagePostHandler
+fission fn create --name restapi-get --executortype newdeploy --maxscale 3 --env go --pkg restapi-go-pkg --entrypoint MessageGetHandler
+```
+
+Create `HTTPTriggers` for the endpoints for the above created functions
+
+```bash
+fission httptrigger create --url /guestbook/messages --method POST --function restapi-post --name restpost
+fission httptrigger create --url "/guestbook/messages/{id:[0-9]+}" --method PUT --function restapi-update --name restupdate
+fission httptrigger create --url "/guestbook/messages/{id:[0-9]+}" --method GET --function restapi-get --name restgetpart
+fission httptrigger create --url "/guestbook/messages/{id:[0-9]+}" --method DELETE --function restapi-delete --name restdelete
+fission httptrigger create --url "/guestbook/messages/" --method GET --function restapi-get --name restget
+```
+
+### Fission Spec
+
+The entire installation can be done using a single block of `fission spec` commands as shown below
+
+```bash
+fission spec init
+fission env create --name go --image fission/go-env-1.16 --builder fission/go-builder-1.16 --spec
+fission pkg create --name restapi-go-pkg --sourcearchive restapi-go-pkg.zip --env go --spec
+fission fn create --name restapi-delete --executortype newdeploy --maxscale 3 --env go --pkg restapi-go-pkg --entrypoint MessageDeleteHandler --spec
+fission fn create --name restapi-update --executortype newdeploy --maxscale 3 --env go --pkg restapi-go-pkg --entrypoint MessageUpdateHandler --spec
+fission fn create --name restapi-post --executortype newdeploy --maxscale 3 --env go --pkg restapi-go-pkg --entrypoint MessagePostHandler --spec
+fission fn create --name restapi-get --executortype newdeploy --maxscale 3 --env go --pkg restapi-go-pkg --entrypoint MessageGetHandler --spec
+fission httptrigger create --url /guestbook/messages --method POST --function restapi-post --spec --name restpost
+fission httptrigger create --url "/guestbook/messages/{id:[0-9]+}" --method PUT --function restapi-update --spec --name restupdate
+fission httptrigger create --url "/guestbook/messages/{id:[0-9]+}" --method GET --function restapi-get --spec --name restgetpart
+fission httptrigger create --url "/guestbook/messages/{id:[0-9]+}" --method DELETE --function restapi-delete --spec --name restdelete
+fission httptrigger create --url "/guestbook/messages/" --method GET --function restapi-get --spec --name restget
+```
+
+It will create a `specs` folder with all the required yaml files post which you can apply the changes
+
+```bash
+fission spec apply
+```
+
+## Testing The Guestbook Application
+
+Since this guestbook application is created using `REST` APIs, we first need to expore the `$FISSION_ROUTER`
+
+```bash
+export FISSION_ROUTER=$(minikube ip):$(kubectl -n fission get svc router -o jsonpath='{...nodePort}')
```

Comment:
I think it would be better to use port forwarding here so that users using other local clusters don't have do any extra steps.
```
export PORT=8889
kubectl --namespace fission port-forward $(kubectl --namespace fission get pod -l svc=router -o name) $PORT:8888
export FISSION_ROUTER=127.0.0.1:$PORT
```

Reply:
Makes sense, update the code snippet.